backend/core/src/main/java/org/sonarsource/sonarlint/core/embedded/server/ShowFixSuggestionRequestHandler.java

```diff
@@ -0,0 +1,376 @@
+    var requestEntityString = EntityUtils.toString(request.getEntity(), "UTF-8");
+    FixSuggestionPayload payload = null;
+    try {
+      payload = new Gson().fromJson(requestEntityString, FixSuggestionPayload.class);
+    } catch (Exception e) {
+      // will be converted to HTTP response later
+      LOG.error("Could not deserialize fix suggestion payload", e);
+    }
+    return payload;
+  }
+
+  @VisibleForTesting
+  public static class ShowFixSuggestionQuery {
+    private final String serverUrl;
+    private final String projectKey;
+    private final String issueKey;
+    private final String branch;
+    @Nullable
+    private final String tokenName;
+    @Nullable
+    private final String tokenValue;
+    @Nullable
+    private final String organizationKey;
+    private final boolean isSonarCloud;
+    private final FixSuggestionPayload fixSuggestion;
+
+    public ShowFixSuggestionQuery(@Nullable String serverUrl, String projectKey, String issueKey, String branch,
+      @Nullable String tokenName, @Nullable String tokenValue, @Nullable String organizationKey, boolean isSonarCloud,
+      FixSuggestionPayload fixSuggestion) {
+      this.serverUrl = serverUrl;
+      this.projectKey = projectKey;
+      this.issueKey = issueKey;
+      this.branch = branch;
+      this.tokenName = tokenName;
+      this.tokenValue = tokenValue;
+      this.organizationKey = organizationKey;
+      this.isSonarCloud = isSonarCloud;
+      this.fixSuggestion = fixSuggestion;
+    }
+
+    public boolean isValid() {
+      return isNotBlank(projectKey) && isNotBlank(issueKey) && isNotBlank(branch)
+        && (isSonarCloud || isNotBlank(serverUrl))
+        && (!isSonarCloud || isNotBlank(organizationKey))
+        && fixSuggestion.isValid() && isTokenValid();
+    }
+
+    /**
+     * Either we get a token combination or we don't get a token combination: There is nothing in between
+     */
+    public boolean isTokenValid() {
+      if (tokenName != null && tokenValue != null) {
+        return isNotEmpty(tokenName) && isNotEmpty(tokenValue);
+      }
+
+      return tokenName == null && tokenValue == null;
+    }
+
+    public String getServerUrl() {
+      return serverUrl;
+    }
+
+    public String getProjectKey() {
+      return projectKey;
+    }
+
+    @Nullable
+    public String getOrganizationKey() {
+      return organizationKey;
+    }
+
+    public String getIssueKey() {
+      return issueKey;
+    }
+
+    public String getBranch() {
+      return branch;
+    }
+
+    @Nullable
+    public String getTokenName() {
+      return tokenName;
+    }
+
+    @Nullable
+    public String getTokenValue() {
+      return tokenValue;
+    }
+
+    public FixSuggestionPayload getFixSuggestion() {
+      return fixSuggestion;
+    }
+  }
+
+  @VisibleForTesting
+  public static class FixSuggestionPayload {
+    private final FileEditPayload fileEdit;
+    private final String suggestionId;
```

Comment:
In tests, an example of a payload looks like UUID. So the field can be not String but UUID.